### PR TITLE
[Backend] New module OpenMetadata-clients With java SDKs

### PR DESCRIPTION
**Pull request by mohitdeuex**
What's working with this :-
1. Can make calls to local running instance of catalog-rest-service without any authorizations
2. Create Clients for all the APIs , currently majority of the APIs received a generic Response object , which has status, message and body , which can be used to read the response of the object.
3. Version Checking between Client and Server.
Work in Progress :-
1. Need to see if we need to create a single distribution package containing all the required dependency for the client.
2. Add the Custom OAuth Interceptors in case of Security Config given in the OpenMetaData configurations.
3. Adding unit test cases for the APIs.
4. Work on optimizing the calls to have a generic API client, for POST , PUT , GETBYID, GETBYFQN , listing individual elements would still require different client i think (Need to explore this).

### Describe your changes :
Created New Module OpenMetadata-clients which will have all the API clients supported by OMD.
Structure to be followed :-
 openmetadata-clients
   - openmetadata-java-client
   - openmetadata-python-client
   - openmetadata-{language}-client

#
### Type of change :

- [x] New feature

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@harshach @sureshms

### CODE REVIEW

**Comment by sureshms** [comment]
@mohitdeuex, which part of this code is automatically generated vs. hand coded?

**Comment by mohitdeuex** [comment]
@sureshms the definitions and Api Client were generated using the swagger , but everything in api/ had to be changed to correct response, since the generated code does not align with the response the server gives . Currently those definition require manual changes, but I believe once we have stub for both server and client generated , we can safely remove the generated ones.
Following are the file that are generated :-
 api/ -> generated but return type changed as per Response.
 gateway/ -> not generated
 auth/ -> generated , trying to use generated OAuth for  tokens refreshing.
 model/ -> not generated
 security/ -> not generated , Dev pending

**Review by harshach — APPROVED** [review]

File: openmetadata-clients/openmetadata-java-client/src/main/java/org/openmetada/restclient/ApiClient.java [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.openmetada.restclient.auth.ApiKeyAuth;
+import org.openmetada.restclient.auth.HttpBasicAuth;
+import org.openmetada.restclient.auth.OAuth;
+import org.apache.oltu.oauth2.client.request.OAuthClientRequest.AuthenticationRequestBuilder;
+import org.apache.oltu.oauth2.client.request.OAuthClientRequest.TokenRequestBuilder;
+import org.threeten.bp.*;
```
Comment (harshach) [inline openmetadata-clients/openmetadata-java-client/src/main/java/org/openmetada/restclient/ApiClient.java:11]: avoid wild-card imports

File: openmetadata-clients/openmetadata-java-client/src/main/java/org/openmetada/restclient/api/DatabaseSchemasApi.java [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+import org.openmetada.restclient.ApiClient;
+import org.openmetada.restclient.EncodingUtils;
+import org.openmetadata.core.util.ResultList;
+import org.openmetadata.catalog.type.EntityHistory;
+import org.openmetadata.catalog.api.data.CreateDatabaseSchema;
+import org.openmetadata.catalog.entity.data.DatabaseSchema;
+
+import java.util.HashMap;
+import java.util.Map;
+import feign.*;
```
Comment (harshach) [inline openmetadata-clients/openmetadata-java-client/src/main/java/org/openmetada/restclient/api/DatabaseSchemasApi.java:12]: avoid wild-card imports

**Comment by mohitdeuex** [comment]
Closing this , we have other PR with generated code

**Pull request closed without merging**